src/main/java/net/reini/rabbitmq/cdi/ConnectionRepository.java

```diff
@@ -0,0 +1,104 @@
+package net.reini.rabbitmq.cdi;
+
+import com.rabbitmq.client.Connection;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+
+/**
+ * <p>
+ * Repository to hold one single connection to each broker
+ * </p>
+ *
+ * <p>
+ * It is recommended by the RabbitMQ documentation (v2.7) to use one single connection within a client and to use one channel for every client thread.
+ * </p>
+ *
+ * @author Patrick Reinhart
+ */
+@ApplicationScoped
+public class ConnectionRepository {
+
+  private Function<ConnectionConfiguration, ConnectionManager> connectionManagerFactoryFunction;
+
+  private final Map<ConnectionConfiguration, ConnectionManager> connectionManagers;
+
+  public ConnectionRepository() {
+    this(ConnectionManager::new);
+  }
+
```

Comment:
reduce blank lines between methods to one line

Reply:
fixed